File: src/Client/realestate.webapp/src/viewModels/realEstate/propertyInventory/propertyInventoryModel.ts
import type { BaseViewModel } from "../../baseViewModel";

export interface PropertyInventoryListViewModel extends BaseViewModel {
  pagesCount: number;
  itemsList: PropertyInventoryListItemViewModel[];
}

export interface PropertyInventoryListItemViewModel extends BaseViewModel {
  structureType: StructureType;
  realEstateTypesList?: string;
  title: string; // required
  price: number;
  totalOfRooms?: string;
  capacity?: string;
  region?: string;
  builder?: string;
  date?: string; 
  editCommand?: string; 
}
export interface PropertyInventoryBaseViewModel extends BaseViewModel {
  structureType: StructureType;
  positionType: PositionType;
 

  startDate?: string | null;
  finishDate?: string | null;
  price: number;
  totalOfRooms?: string | null;
  capacity?: string | null;
  currency: Currency;
  regionId: string | null; 
  builderId: string | null; 

  title: string; // required
  description?: string;

  title_Fa: string; // required
  description_Fa?: string;

  title_Ar: string; // required
  description_Ar?: string;

  shortDescription: string; // required
  shortDescription_Fa: string; // required
  shortDescription_Ar: string; // required

  selectedTagIdsList_En?: string[]; // Guid[] → string[]
  selectedTagIdsList_Fa?: string[];
  selectedTagIdsList_Ar?: string[];

  selectedTypesList?: RealEstateType[];
  selectedFeaturesList?: FeatureType[];

  prepayment?: string;
  duringProjectPayment?: string;
  handoverPayment?: string;
  afterHandoverPayment?: string;
  brochureLink?: string;
}
export interface PropertyInventoryViewModel extends PropertyInventoryBaseViewModel {
  attachments?: FileUploadViewModel;
  
  tagsList_En?: SelectListItem[];
  tagsList_Fa?: SelectListItem[];
  tagsList_Ar?: SelectListItem[];
  
  structureTypesList?: SelectListItem[];
  realEstateTypesList?: SelectListItem[];
  currenciesList?: SelectListItem[];
  regionsList?: SelectListItem[];
  buildersList?: SelectListItem[];
  featuresList?: SelectListItem[];
  positionTypesList?: SelectListItem[];
}
export interface SelectListItem {
  value: string;
  text: string;
  selected?: boolean;
}
export interface FileUploadViewModel {
  fileName: string;
  filePath: string;
  size?: number;
}
export interface PropertyInventoryFilterViewModel {
  pageNumber: number;
  pageSize: number;
  language: Language;
  structureType?: StructureType;
  realEstateType?: RealEstateType;
  startDate?: string;
  finishDate?: string;
  price_From?: number;
  price_To?: number;
  regionId?: string | null; 
  builderId?: string | null; 
  title?: string | undefined | null;
  orderBy: PropertyInventoryOrderType;
  languagesList?: SelectListItem[];
  structureTypesList?: SelectListItem[];
  realEstateTypesList?: SelectListItem[];
  regionsList?: SelectListItem[];
  buildersList?: SelectListItem[];
}
//enums

// StructureType
export type StructureType = 0 | 1;

export const StructureTypeValues = {
  Ready: 0 as StructureType,
  Inprogress: 1 as StructureType,
};

// Currency
export type Currency = 0 | 1;

export const CurrencyTypeValues = {
  AED: 0 as Currency,
  USD: 1 as Currency,
};

export type PositionType = 0 | 1;

export const PositionTypeValues = {
  Freehold: 0 as PositionType,
  Leasehold: 1 as PositionType,
};

export type RealEstateType = 0 | 1 | 2 | 3 | 4;

export const RealEstateTypeValues = {
  Apartment: 0 as RealEstateType,
  Villa: 1 as RealEstateType,
  TownHouse: 2 as RealEstateType,
  PentHouse: 3 as RealEstateType,
  Duplex: 4 as RealEstateType,
};

// FeatureType.ts
export type FeatureType =
  | 0
  | 1
  | 2
  | 3
  | 4
  | 5
  | 6
  | 7
  | 8
  | 9
  | 10
  | 11
  | 12;

export const FeatureTypeValues = {
  SmartHome: 0 as FeatureType,
  ChildrenPlayground: 1 as FeatureType,
  Pool: 2 as FeatureType,
  Gym: 3 as FeatureType,
  Security: 4 as FeatureType,
  GreenAreas: 5 as FeatureType,
  JoggingTracks: 6 as FeatureType,
  PadelCourt: 7 as FeatureType,
  OutdoorDiningArea: 8 as FeatureType,
  BBQArea: 9 as FeatureType,
  OutdoorLoungeSpace: 10 as FeatureType,
  Clubhouse: 11 as FeatureType,
  PremiumServices: 12 as FeatureType,
};

export type PropertyInventoryOrderType = 0 | 1 | 2 | 3 | 4 | 5;

export const PropertyInventoryOrderTypeValues = {
  Default: 0 as PropertyInventoryOrderType,
  Price_Low_To_High: 1 as PropertyInventoryOrderType,
  Price_High_To_Low: 2 as PropertyInventoryOrderType,
  Date_New_To_Old: 3 as PropertyInventoryOrderType,
  Date_Old_To_New: 4 as PropertyInventoryOrderType,
  CreateDate: 5 as PropertyInventoryOrderType,
};

export type Language = 0 | 1 | 2;

export const LanguageValues = {
  Persian: 0 as Language,
  English: 1 as Language,
  Arabic: 2 as Language,
};

File: src/Services/RealEstate/Core/RealEstate.Application/DTOs/RealEstate/PropertyInventoryModel.cs
using System.ComponentModel.DataAnnotations;
using RealEstate.Application.DTOs.Common;
using Shared.Enums;

namespace RealEstate.Application.DTOs.RealEstate
{
    public class PropertyInventoryListDTO : BaseModel
	{
		public int PagesCount { get; set; }
		public required List<PropertyInventoryListItemDTO> ItemsList { get; set; }
	}
	public class PropertyInventoryListItemDTO : BaseModel
	{	
		public StructureType StructureType { get; set; }
		public string? RealEstateTypesList { get; set; }
		public  string? Title { get; set; }
        public  string? UrlTitle { get; set; }
		public string? Description { get; set; }
        public string? StartDate { get; set; }
		public string? FinishDate { get; set; }
		public int Price { get; set; }
		public string? TotalOfRooms { get; set; }
		public string? Capacity { get; set; }
		public Currency Currency { get; set; }
		public string? Region { get; set; }
		public string? Builder { get; set; }
		public string? MainFileName { get; set; }
        public string? MainFilePath { get; set; } 
    }
	public class PropertyInventoryDTO:BaseModel
	{
		public StructureType StructureType { get; set; }
        public PositionType PositionType { get; set; }
        public string? AfterHandoverPayment { get; set; }
        public string? StartDate { get; set; }
		public string? FinishDate { get; set; }
		public int Price { get; set; }
		public string? TotalOfRooms { get; set; }
		public string? Capacity { get; set; }
		public Currency Currency { get; set; }
		public Guid RegionId { get; set; }
		public Guid BuilderId { get; set; }
		[Required]
		public string? Title { get; set; }
        public string? Description { get; set; }
		[Required]
		public string? Title_Fa { get; set; }
		public string? Description_Fa { get; set; }
		[Required]
		public string? Title_Ar { get; set; }
		public string? Description_Ar { get; set; }
        public string? ShortDescription { get; set; }
        public string? ShortDescription_Fa { get; set; }
        public string? ShortDescription_Ar { get; set; }
        public List<FileUploadViewModelPost>? FilesList { get; set; }
		public List<Guid>? SelectedTagIdsList_En { get; set; }
		public List<Guid>? SelectedTagIdsList_Fa { get; set; }
		public List<Guid>? SelectedTagIdsList_Ar { get; set; }
        public List<RealEstateType>? SelectedTypesList { get; set; }       
        public List<FeatureType>? SelectedFeaturesList { get; set; }      
        public string? Prepayment { get; set; }       
        public string? DuringProjectPayment { get; set; }      
        public string? HandoverPayment { get; set; }
        public string? BrochureLink { get; set; }
    }
    public class PropertyInventoryFromFilesDTO: PropertyInventoryDTO
    {
        public string? Region { get; set; }
        public string? Builder { get; set; }
    }
    public class UploadFileResultDTO
	{
        public int Code { get; set; } 
        public List<FileUploadViewModelPost>? AddedFilesList { get; set; }
        public List<FileUploadViewModelPost>? DeletedFilesList { get; set; }
    }
    public class PropertyInventoryEndUserDTO
    {
        public StructureType StructureType { get; set; }
        //public RealEstateType RealEstateType { get; set; }
        public string? StartDate { get; set; }
        public string? FinishDate { get; set; }
        public int Price { get; set; }
        public PositionType PositionType { get; set; }
        public string? Builder { get; set; }
        public string? AmenitiesList { get; set; }
        public string? TotalOfRooms { get; set; }
        public string? Capacity { get; set; }
        public Currency Currency { get; set; }
        public string? Region { get; set; }
        [Required]
        public string? Title { get; set; }
        public string? Description { get; set; }
        public string? ShortDescription { get; set; }
        public string? Prepayment { get; set; }
        public string? DuringProjectPayment { get; set; }
        public string? HandoverPayment { get; set; }
        public string? AfterHandoverPayment { get; set; }
        public string? BrochureLink { get; set; }
        public List<FileUploadViewModelPost>? FilesList { get; set; }
        public List<string>? SelectedTagsList { get; set; }
		public List<PropertyInventoryRecommendationDTO>? RecommendationList { get; set; } 
    }
    public class PropertyInventoryRecommendationDTO
    {
        public int Price { get; set; }
        public Currency Currency { get; set; }
        public Guid PropertyInventoryId { get; set; }
        public string? Title { get; set; }
		public string? UrlTitle { get; set; }
		public string? FileName { get; set; }
    }
}
